src/test/java/com/juniorstart/juniorstart/service/EmploymentHistoryServiceTest.java

```diff
@@ -0,0 +1,160 @@
+package com.juniorstart.juniorstart.service;
+
+import com.juniorstart.juniorstart.exception.BadRequestException;
+import com.juniorstart.juniorstart.exception.ResourceNotFoundException;
+import com.juniorstart.juniorstart.model.*;
+import com.juniorstart.juniorstart.repository.EmploymentHistoryRepository;
+import com.juniorstart.juniorstart.repository.UserProfileRepository;
+import com.juniorstart.juniorstart.security.UserPrincipal;
+
+import org.junit.jupiter.api.*;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.*;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import java.sql.Date;
+import java.util.Optional;
+import java.util.UUID;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Mockito.*;
+
+@ExtendWith(MockitoExtension.class)
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+public class EmploymentHistoryServiceTest {
+
+    @Spy
+    UserProfileRepository userProfileRepository;
+    @Spy
+    EmploymentHistoryRepository employmentHistoryRepository;
+    @InjectMocks
+    EmploymentHistoryService employmentHistoryService;
+
+    User user;
+    UserProfile userProfile;
+    UserPrincipal userPrincipal;
+    EmploymentHistory employmentHistory;
+
+    Date startDate = Date.valueOf("2020-10-01");
+    Date endDate = Date.valueOf("2020-10-30");
+
+    UUID example_UUID = UUID.randomUUID();
+
+    @BeforeEach
+    public void setUp() {
+        MockitoAnnotations.initMocks(this);
+
+        user = User.builder()
+                .privateId(example_UUID)
+                .publicId(10L)
+                .name("Test")
+                .email("test2@test.com")
+                .emailVerified(true)
+                .password("Test%123")
+                .provider(AuthProvider.google).build();
+
+        userPrincipal = UserPrincipal.create(user);
+
+
+        userProfile = UserProfile.builder()
+                .privateId(example_UUID)
+                .user(user).build();
+
+        employmentHistory = EmploymentHistory.builder().id(0).dateStartOfEmployment(startDate)
+                .companyName("companyName").tasksAtWork("Task1").isCurrentEmployment(false).build();
+    }
+
+    @Test
+    public void should_addEmploymentHistory() {
+
+        employmentHistory.setDateEndOfEmployment(endDate);
+
+        Mockito.when(userProfileRepository.findByPrivateId(ArgumentMatchers.any(UUID.class))).thenReturn(Optional.of(userProfile));
+
+        ResponseEntity<?> status = employmentHistoryService.addEmploymentHistory(employmentHistory, userPrincipal);
+
+        verify(userProfileRepository, times(1)).save(userProfile);
+        assertThat(status.getStatusCode()).isEqualTo(HttpStatus.OK);
+    }
+
```

Comment:
Zaznaczę jeden test ale odnosi się do wszystkich.
1. Eleganckie testy, czytelne i dobrze zrobione. Podoba mi się :D Jedynie drobne uwagi
2. Mockito.when zamień na when. Jest statyczne żeby nie trzeba było pisać mockito
3. @DisplayName warto jest wstawić, bo przy robieniu testów zamiast nazwy np: should_addRecentEmployer możesz zrobić Should add recent employer. Dużo bardziej czytelne to jest.
4. Jak chcesz żeby jeszcze bardziej czytelny kod testów był możesz użyć komentarzy //given // when // then

Reply:
2- Tak, ale czytając/ przeglądając kod na szybko lepiej wiedzieć, że odnosimy się wprost do Mockito.
3- Właśnie specjalnie starałem się nazwać klasy, żeby świadczyły o tym jaki test wykonują, żeby pominąć tą adnotację.
4 - Widziałem , że tak ludzie to stosują, ale wygodniej dla mnie, gdy po prostu bez pisania //given //when /then 
     rozbiję to na kod (poprzez entery) Chyba, że sytuacja tego wymaga to jestem elastyczny, ale wole nie :D

Reply:
1."lepiej wiedzieć, że odnosimy się wprost do Mockito." osoba która programuje to wie. To jest standardowe narzędzie. Poza tym masz od tego do góry listę importów żeby wiedzieć skąd się dana klasa wzięła.
2. ale nadal podłogi i tekst wielbłądzi na wyświetlaczu wyglądają dużo gorzej. Jak będziesz budować klase przez mavena i na konsoli CI się wyświetlą informacje to lepiej jest przeczytać "Get unauthorized test" niż "getUnauthorizedTest"
3. Jak uważasz

Reply:
Mockito zmienię, w sprawie nazwy testów, niech się inni jeszcze wypowiedzą
@hosu794  @Ejden